core/runtime/src/main/java/io/quarkus/runtime/logging/LogMetricsHandlerRecorder.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.runtime.logging;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.logging.Handler;
+
+import org.jboss.logmanager.Level;
+
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.runtime.metrics.MetricsFactory;
+
+@Recorder
+public class LogMetricsHandlerRecorder {
+
+    static final String METRIC_NAME = "log.total";
+
+    static final String METRIC_DESCRIPTION = "Number of log events";
+
+    static final List<Level> STANDARD_LEVELS = Arrays.asList(Level.FATAL, Level.ERROR, Level.WARN, Level.INFO, Level.DEBUG,
+            Level.TRACE);
+
+    Map<Level, AtomicLong> logCounters = new HashMap<>();
+
+    /**
+     * Counter for non-standard log levels
+     */
+    AtomicLong defaultLogCounter = new AtomicLong(0);
+
+    public Consumer<MetricsFactory> registerMetrics() {
+        return new Consumer<MetricsFactory>() {
+            @Override
+            public void accept(MetricsFactory metricsFactory) {
+                metricsFactory.builder(METRIC_NAME).description(METRIC_DESCRIPTION)
+                        .buildCounter(defaultLogCounter::get);
+
+                for (Level level : STANDARD_LEVELS) {
+                    AtomicLong counter = new AtomicLong(0);
+                    metricsFactory.builder(METRIC_NAME).description(METRIC_DESCRIPTION).tag("level", level.getName())
+                            .buildCounter(counter::get);
+                    logCounters.put(level, counter);
```

Comment:
I noticed a race condition in bootstrap: the log handler gets called with few INFO/DEBUG events (~15 with MetricsTestCase) before the metrics are themselves registered.
=> I'll pre-fill the Map in the handler itself to solve this.